sunpy/instr/lyra.py

```diff
@@ -1,123 +1,581 @@
+
+    force_use_local_lytaf : bool
+        Ensures current local version of lytaf files are not replaced by
+        up-to-date online versions even if current local lytaf files do not
+        cover entire input time range etc.
+        Default=False
+
+    Returns
+    -------
+    lc_new : LYRALightCurve
+        copy of input LYRALightCurve with periods corresponding to artifacts
+        removed.
+
+    artifact_status : (optional) dictionary
+        List of 4 variables containing information on what artifacts were
+        found, removed, etc. from the time series.
+        artifact_status["lytaf"] = artifacts found : numpy recarray
+            The full LYRA annotation file for the time series time range
+            output by get_lytaf_events().
+        artifact_status["removed"] = artifacts removed : numpy recarray
+            Artifacts which were found and removed from from time series.
+        artifact_status["not_removed"] = artifacts found but not removed :
+              numpy recarray
+            Artifacts which were found but not removed as they were not
+            included when user defined artifacts kwarg.
+        artifact_status["not_found"] = artifacts not found : list of strings
+            Artifacts listed to be removed by user when defining artifacts
+            kwarg which were not found in time series time range.
+
+    References
+    ----------
+    [1] http://proba2.oma.be/data/TARDIS
+
+    Examples
+    --------
+    Remove LARs (Large Angle Rotations) from LYRALightCurve for 4-Dec-2014.
+        >>> lc = sunpy.lightcurve.LYRALightCurve.create("2014-12-02")
+        >>> lc_nolars = remove_artifacts_from_lyralightcurve(lc, artifacts=["LAR"])
+
+    To also retrieve information on the artifacts during that day...
+        >>> lc_nolars, artifact_status = remove_artifacts_from_lyralightcurve(
+                lc, artifacts=["LAR"], return_artifacts=True)
+
+    """
+    # Check that input argument is of correct type
+    if not isinstance(lc, lightcurve.LightCurve):
+        raise TypeError("lc must be a LightCurve object.")
+    # Remove artifacts from time series
+    data_columns = lc.data.columns
+    time, channels, artifact_status = _remove_lytaf_events(
+        lc.data.index,
+        channels=[np.asanyarray(lc.data[col]) for col in data_columns],
+        artifacts=artifacts, return_artifacts=True, lytaf_path=lytaf_path,
+        force_use_local_lytaf=force_use_local_lytaf)
+    # Create new copy copy of lightcurve and replace data with
+    # artifact-free time series.
+    lc_new = copy.deepcopy(lc)
+    lc_new.data = pandas.DataFrame(
+        index=time, data=dict((col, channels[i])
+                              for i, col in enumerate(data_columns)))
+    if return_artifacts:
+        return lc_new, artifact_status
+    else:
+        return lc_new
+
+def _remove_lytaf_events(time, channels=None, artifacts=[],
+                         return_artifacts=False, fitsfile=None,
```

Comment:
kwargs lytaf_path and artifacts should be None and then defined to default inside function.  Same goes for all other functions.